File: java/ciudadanias-europeas/src/main/java/ar/com/phidalgo/entity/Tramite.java
package ar.com.phidalgo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "tramite")
@Getter
@Setter
public class Tramite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_tramite")
    private Long id;
    @OneToOne
    @PrimaryKeyJoinColumn
    private Consulado consulado;
    @OneToOne
    @PrimaryKeyJoinColumn
    private TipoTramite tipoTramite;
    private Float importe;
}

File: angular/mi-proyecto-angular/src/app/modelos/Tramite.ts
import { Consulado } from "./Consulado";
import { TipoTramite } from "./TipoTramite";

export interface Tramite {
  id?: number;
  consulado: Consulado | null;
  tipoTramite: TipoTramite | null;
  importe: number | null;
}